src/modules/others/timer.cpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file timer.cpp
+ * @author Aleksei Gor (https://github.com/AlekseiGor)
+ * @brief Timer
+ * @version 0.1
+ * @date 2024-12-07
+ */
+
+#include "timer.h"
+#include "core/display.h"
+#include "modules/others/audio.h"
+
+#define DELAY_VALUE 150
+
+Timer::Timer() {
+    setup();
+}
+
+void Timer::loop() {
+    unsigned long startMillis = millis();
+    unsigned long currentMillis;
+    unsigned long elapsedMillis;
+
+    char timeString[9];
+
+    tft.fillScreen(bruceConfig.bgColor);
+
+    for (;;) {
+        currentMillis = millis();
+        elapsedMillis = currentMillis - startMillis;
+
+        unsigned long remainingMillis = duration - elapsedMillis;
+
+        int seconds = (remainingMillis / 1000) % 60 + 1;
+        int minutes = (remainingMillis / 60000) % 60;
+        int hours = (remainingMillis / 3600000);
+
+        snprintf(timeString, sizeof(timeString), "%02d:%02d:%02d", hours, minutes, seconds);
+
+        tft.setTextColor(bruceConfig.priColor, bruceConfig.bgColor);
+        drawMainBorder(false);
+        tft.setTextSize(4);
+        tft.drawCentreString(timeString, WIDTH / 2, HEIGHT / 2 - 13, 1);
+
+        if (checkEscPress()) {
+            duration = 0;
+            tft.fillScreen(bruceConfig.bgColor);
+            returnToMenu = true;
+            break;
+        }
+
+        if (elapsedMillis >= duration) {
+            tft.fillScreen(bruceConfig.bgColor);
+            _tone(2000, 1000);
+            returnToMenu = true;
+            break;
+        }
+
+        delay(DELAY_VALUE);
+    }
+}
+
+void Timer::setup() {
+    int hours = 0;
+    int minutes = 0;
+    int seconds = 0;
+    int settingMode = 0;
+    int underlineHeight = HEIGHT / 3 * 2;
+    int underlightWidth = WIDTH / 5;
```

Comment:
should this be underlineWidth instead of underlightWidth?

Reply:
indeed, it was a typo
done